docs/deprecation-policy.md

```diff
@@ -0,0 +1,186 @@
+---
+assignees:
+- bgrant0607
+- lavalamp
+- thockin
+
+---
+
+# Kubernetes Deprecation Policy
+
+Kubernetes is a large system with many components and many contributors.  As
```

Comment:
Try to avoid "we" constructs.

"Kubernetes is a large system with many components and contributors. As with any such software, the feature set evolves over time, and sometimes a feature may need to be removed. This could include an API, a flag, or even an entire feature. To avoid making breaking changes for existing users, Kubernetes is instituting a *deprecation policy* for aspects of the system that are slated to be removed."